### ISSUE TITLE
SMS protocol : avoid sending new WO twice

### ISSUE THREAD

**Issue by estellecomment** [medic/medic-webapp#3408]
From #3073 :
Modify medic-gateway to call medic-api and change the status of received messages to received-by-gateway after they have been successfully written to the database. This will stop them being sent again next query and reduce the bandwidth requirements.

This will avoid step 4 in [the current protocol](https://github.com/medic/medic-docs/blob/e4483cbae3b619f1db8f1990f43cd25905e7b09d/user/message-states.md)

To do it, have gateway set the `needs forwarding` flag to `true` when storing a new message in DB.

**Comment by alxndrsn** [comment]
@estellecomment It look like you've done most of the work for this.  Is it in progress, or do you want me to finish it off?

**Comment by estellecomment** [comment]
I haven't given up on it yet :) 
I unassigned myself because I'm doing something else for now, and I don't know how much of a hurry you're in. If it's urgent you can finish it, otherwise I will today/tomorrow.

**Comment by estellecomment** [comment]
@alxndrsn Can you review?

**Comment by alxndrsn** [comment]
PR test failures are things like this:
```
medic.gateway.alert.SettingsDialogActivityTest > generic_shouldDisplayCorrectFields[test(AVD) - 5.0.2] FAILED 

	java.lang.RuntimeException: Waited for the root of the view hierarchy to have window focus and not be requesting layout for over 10 seconds. If you specified a non default root matcher, it may be picking a root that never takes focus. Otherwise, something is seriously wrong. Selected Root:
```

The espresso tests intermittently fail on Travis.  I will re-run and see if they pass the second time.

**Comment by garethbowen** [comment]
@alxndrsn What's the status of this? Did it fail AT?

**Comment by estellecomment** [comment]
nah, I merged it but @alxndrsn hadn't reviewed it so reverted it. 
It got split into two PRs : 
 - code : https://github.com/medic/medic-gateway/pull/107 is breaking, I have to figure out why
 - unrelated tests : https://github.com/medic/medic-gateway/pull/106 is supposed to be all tests but apparently it's not, so gotta sort that out.

I have higher priority stuff that keeps happening, but it's still on my list.

**Comment by garethbowen** [comment]
@alxndrsn Would you mind finishing this off while @estellecomment is away?

**Comment by alxndrsn** [comment]
I didn't have time to refamiliarise myself with the original ticket, and the information about motivation here seems sparse.  I would like to fully consider whether this is useful, in line with discussions about the protocol and recent work on the server-side by @SCdF.  But if we _do_ want to achieve this, PR at https://github.com/medic/medic-gateway/pull/113

**Comment by garethbowen** [comment]
@estellecomment Should we finish this or close it? It's been hanging around for ages...

**Comment by garethbowen** [comment]
Ok... This isn't getting done. Kicking it out of the milestone.

**Comment by dianabarsan** [comment]
@alxndrsn what do you think? Worth reviving?
